src/main/java/io/github/jhipster/config/apidoc/SwaggerAutoConfiguration.java

```diff
@@ -0,0 +1,185 @@
+import org.springframework.util.StopWatch;
+import org.springframework.util.StringUtils;
+import org.springframework.web.servlet.DispatcherServlet;
+import springfox.bean.validators.configuration.BeanValidatorPluginsConfiguration;
+import springfox.documentation.schema.AlternateTypeRule;
+import springfox.documentation.service.ApiInfo;
+import springfox.documentation.service.Contact;
+import springfox.documentation.spi.DocumentationType;
+import springfox.documentation.spring.web.plugins.Docket;
+import springfox.documentation.swagger2.annotations.EnableSwagger2;
+
+import javax.servlet.Servlet;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+
+import static io.github.jhipster.config.JHipsterConstants.SPRING_PROFILE_SWAGGER;
+import static springfox.documentation.builders.PathSelectors.regex;
+
+/**
+ * Springfox Swagger configuration.
+ * <p>
+ * Warning! When having a lot of REST endpoints, Springfox can become a performance issue. In that case, you can use a
+ * specific Spring profile for this class, so that only front-end developers have access to the Swagger view.
+ */
+@Configuration
+@ConditionalOnWebApplication
+@ConditionalOnClass({
+    ApiInfo.class,
+    BeanValidatorPluginsConfiguration.class,
+    Servlet.class,
+    DispatcherServlet.class
+})
+@Profile(SPRING_PROFILE_SWAGGER)
+@AutoConfigureAfter(JHipsterProperties.class)
+@EnableSwagger2
+@Import(BeanValidatorPluginsConfiguration.class)
+public class SwaggerAutoConfiguration {
+
+    public static final String STARTING_MESSAGE = "Starting Swagger";
+    public static final String STARTED_MESSAGE = "Started Swagger in {} ms";
+    public static final String MANAGEMENT_TITLE_SUFFIX = "Management API";
+    public static final String MANAGEMENT_GROUP_NAME = "management";
+    public static final String MANAGEMENT_DESCRIPTION = "Management endpoints documentation";
+
+    private final Logger log = LoggerFactory.getLogger(SwaggerAutoConfiguration.class);
+
+    private final JHipsterProperties.Swagger properties;
+
+    public SwaggerAutoConfiguration(JHipsterProperties jHipsterProperties) {
+        this.properties = jHipsterProperties.getSwagger();
+    }
+
+    @Bean
+    public BuildInSwaggerCustomizer buildInSwaggerCustomizer(
+        ObjectProvider<AlternateTypeRule[]> alternateTypeRulesProviders) {
+        AlternateTypeRule[] alternateTypeRules = alternateTypeRulesProviders.getIfAvailable();
+        if (alternateTypeRules == null) {
+            alternateTypeRules = new AlternateTypeRule[]{};
+        }
+        return new BuildInSwaggerCustomizer(alternateTypeRules);
+    }
+
+    /**
+     * Springfox configuration for the API Swagger docs.
+     *
+     * @return the Swagger Springfox configuration
+     */
+    @Bean
+    @ConditionalOnMissingBean(name = "swaggerSpringfoxApiDocket")
+    public Docket swaggerSpringfoxApiDocket(ObjectProvider<SwaggerCustomizer[]> swaggerCustomizerProviders) {
```

Comment:
It should be possible to inject a list of SwaggerCustomizers directly

Reply:
Fixed